### PR TITLE
unit tests and bugs 

### PR DESCRIPTION
**Pull request by Shmuel-columbia**

### CODE REVIEW

File: lightning_pose_app/backend/extract_frames.py [hunk 1]
```diff
@@ -465,14 +469,14 @@ def zip_annotated_images(labeled_data_check_path):
     return zip_buffer
 
 
-@st.cache_data(show_spinner=False)
-def load_image(image_path: str) -> Image:
-    return Image.open(image_path)
+# @st.cache_data(show_spinner=False)
+# def load_image(image_path: str) -> Image:
+#     return Image.open(image_path)
```
Comment (themattinthehatt) [inline lightning_pose_app/backend/extract_frames.py:474]: can delete this commented out function if no longer needed

File: lightning_pose_app/ui/project.py [hunk 2]
```diff
@@ -434,11 +438,47 @@ def find_top_level_dir(initial_path, target_file_name=None, target_dir=None):
...
+            #         _logger.info(
+            #             f"Deleted zipped project file at {self.st_upload_existing_project_slp}"
+            #         )
+            #     except Exception as e:
+            #         _logger.error(f"Failed to delete slp project file: {e}")
+            # else:
+            #     _logger.error(
+            #         f"Slp project file at \
+            #         {self.st_upload_existing_project_slp} does not exist"
+            #     )
```
Comment (themattinthehatt) [inline lightning_pose_app/ui/project.py:481]: should verify this works and include it in the PR

File: lightning_pose_app/ui/train_infer.py [hunk 3]
```diff
@@ -1158,7 +1158,7 @@ def _render_streamlit_fn(state: AppState):
 
             st.header("Create an ensemble of models")
             selected_models = st.multiselect(
-               "Select models for ensembling",
+               "Select models for ensembling", # noqa
```
Comment (themattinthehatt) [inline lightning_pose_app/ui/train_infer.py:1161]: why is #noqa here?

File: tests/backend/test_backend_extract_frames.py [hunk 4]
```diff
@@ -258,3 +259,73 @@ def test_find_contextual_frames():
...
+    ]
+    data = [[scorer_data[i]] + fixed_values[i] for i in range(2)]
+    all_data = header_data + data
+
+    with open(csv_path, 'w', newline='') as file:
+        writer = csv.writer(file)
+        writer.writerows(all_data)
+
+    result_dict = {}
+    for row in data:
```
Comment (themattinthehatt) [inline tests/backend/test_backend_extract_frames.py:289]: this is great!

File: tests/backend/test_backend_extract_frames.py [hunk 5]
```diff
@@ -258,3 +259,73 @@ def test_find_contextual_frames():
...
+            'frame_full_path': os.path.join(tmpdir, frame_path),
+            'video': 'test_vid',
+            'frame_number': int(frame_path.split('img')[-1].split('.')[0]),
+            'bodyparts': bodyparts_dict
+        }
+
+    return csv_path, result_dict
+
+
+def test_convert_csv_to_dict(tmpdir):
```
Comment (themattinthehatt) [inline tests/backend/test_backend_extract_frames.py:307]: very nice, much cleaner now

**Review by themattinthehatt — CHANGES_REQUESTED** [review]

File: lightning_pose_app/ui/train_infer.py (hunk 3 above)
Comment (Shmuel-columbia) [reply lightning_pose_app/ui/train_infer.py:1161]: removed

**Review by themattinthehatt — APPROVED** [review]

**Pull request merged**